tests/tests_unit/approval_client.py

```diff
@@ -555,23 +555,44 @@ def return_value(*args, **kwargs):
         method = available_post_methods[mock_method]
         return method
 
-    def dump(self) -> dict[str, Any]:
+    def dump(self, sort: bool = True) -> dict[str, Any]:
         """This returns a dictionary with all the resources that have been created and deleted.
 
+        The sorting is useful in snapshot testing, as it makes for a consistent output. If you want to check the order
```

Comment:
When we do API calls, the order of the calls is important. Don't you remove the ability to test that calls are executed in the correct order?

Reply:
No, I enable it with this PR. 

For snapshot testing you want to sort all resources as you do not care about the order the calls are made, but to test the deploy order I needed to enable checking the order of the files.